leakcanary-android-core/src/main/java/leakcanary/internal/activity/screen/HeapAnalysisListScreen.kt

```diff
@@ -70,10 +65,20 @@ internal class HeapAnalysisListScreen : Screen() {
               view.context, projection.createdAtTimeMillis,
               DateUtils.FORMAT_SHOW_TIME or DateUtils.FORMAT_SHOW_DATE
           )
-
-          titleView.text = "$index. " + (projection.exceptionSummary
-          // TODO Handle singular
-              ?: "${projection.retainedInstanceCount} retained instances")
+          titleView.text =
+          when {
+            projection.exceptionSummary != null ->
+              resources.getString(
+                  R.string.leak_canary_heap_analysis_list_exception_summary, index,
+                  projection.exceptionSummary
+              )
+            else -> String.format(
+                resources.getQuantityText(
+                    R.plurals.leak_canary_heap_analysis_list_retained_instances,
+                    projection.retainedInstanceCount
+                ).toString(), projection.retainedInstanceCount
```

Comment:
This isn't super easy to follow, with the plural on one side wrapped in a string.format